mllib/src/main/scala/org/apache/spark/ml/regression/GeneralizedLinearRegression.scala

```diff
@@ -313,7 +313,11 @@ class GeneralizedLinearRegression @Since("2.0.0") (@Since("2.0.0") override val
    * @group setParam
    */
   @Since("2.0.0")
-  def setSolver(value: String): this.type = set(solver, value)
+  def setSolver(value: String): this.type = {
+    require("irls" == value.toLowerCase(Locale.ROOT),
+      s"Solver $value was not supported. Supported options: irls")
+    set(solver, value)
+  }
```

Comment:
Actually we can't do this, since MLlib supports set params via other entrances. Currently we can leave as it is, until we resolved #16028.